### PR TITLE
Add CI job running under macOS to test the build there too

### PR DESCRIPTION
**Pull request by vadz**
Move almost all the contents of the existing ci/install.sh to
install-Linux.sh and a separate install-Darwin.sh for Mac as this seems
cleaner than dealing with all of them in the same file.

### CODE REVIEW

**Comment by vadz** [comment]
This should be merged once #77 is, as it crashes now, presumably due to the same problem (but to be confirmed).

**Comment by vadz** [comment]
Actually the failures here are unrelated to #77 and are rather due to the problem with catching `xml::exception` from xmlwrapp shared library in the tests due to the fact that it wasn't exported and this is fixed by #79, so once that is merged, I'll rebase and merge this one too.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -67,7 +72,14 @@ jobs:
             echo $1=$2 >> $GITHUB_ENV
           }
 
-          set_env_var PATH /usr/lib/ccache:$PATH
+          case `uname` in
+              Darwin) ccache_path=/usr/local/opt/ccache/libexec ;;
```
Comment (vslavik) [inline .github/workflows/ci.yml:76]: Assuming this path is for Homebrew (looks that way), can you add `/opt/homebrew/opt/ccache/libexec` too? That's the location used by Homebrew on ARM Macs.

Comment (vslavik) [reply .github/workflows/ci.yml:76]: Sorry, momentarily forgot this is for CI (which is unlikely to have ARM anytime soon), so please ignore that.

**Comment by vslavik** [comment]
Looks good to me, thanks.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (vadz) [reply .github/workflows/ci.yml:76]: I hadn't even realized that the path was different for ARM Macs (in spite of using one myself...), but I agree that it's not that important for now, as they're not used for the CI, so I've created https://github.com/hendrikmuhs/ccache-action/issues/101 to maybe improve this in the future, thanks for pointing this out!

**Comment by vadz** [comment]
I'm going to rebase this on #79 when it's merged to check that everything is correct before merging, as the 2 currently failing builds are due to the problems fixed there.

**Pull request merged**